[macOS] Add vector-based version of resizer

lilyspiniolas opened the pull request:
#### 50dbd547b8a226a9da4a573772d5ae2d38fb94ed
<pre>
[macOS] Add vector-based version of resizer
<a href="https://bugs.webkit.org/show_bug.cgi?id=289806">https://bugs.webkit.org/show_bug.cgi?id=289806</a>
<a href="https://rdar.apple.com/147062251">rdar://147062251</a>

Reviewed by Abrar Rahman Protyasha and Aditya Keerthi.

Added support for a vector-based version of the resizer. Moved
codepath for drawing the resizer image and resizer frame to RenderTheme
in order to account for platform differences.

* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::paintResizer):
(WebCore::RenderLayerScrollableArea::drawPlatformResizerImage): Deleted.
* Source/WebCore/rendering/RenderLayerScrollableArea.h:
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::paintPlatformResizer):
(WebCore::RenderTheme::paintPlatformResizerFrame):
* Source/WebCore/rendering/RenderTheme.h:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.h:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::RenderThemeCocoa::paintPlatformResizer):
(WebCore::RenderThemeCocoa::paintPlatformResizerFrame):

Canonical link: <a href="https://commits.webkit.org/292298@main">https://commits.webkit.org/292298@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3349983768db51a3a6b68a146b779dfeb65cadcc

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/95281 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/14881 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/4739 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/100325 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45782 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/97329 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/15169 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/23312 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/72656 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29931 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/98284 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/11329 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/86003 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52988 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/94773 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/11043 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/3744 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/45121 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/87952 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/81218 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/3836 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/102363 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/93904 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/22329 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/16271 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/81653 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/94857 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22577 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/82023 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/81051 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20357 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25625 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/3020 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/15582 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/22299 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/27441 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/116592 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/21958 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/25432 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23697 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
```diff
@@ -1579,20 +1546,11 @@ void RenderLayerScrollableArea::paintResizer(GraphicsContext& context, const Lay
-        context.setFillColor(Color::transparentBlack);
-        context.drawRect(snappedIntRect(largerCorner));
-    }
+    RenderTheme::singleton().paintPlatformResizer(renderer, context, resizerAbsRect);
```
```suggestion
    renderer.theme().paintPlatformResizer(renderer, context, resizerAbsRect);
```

pxlcoder commented on Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
```diff
@@ -1579,20 +1546,11 @@ void RenderLayerScrollableArea::paintResizer(GraphicsContext& context, const Lay
+
+    // Draw a frame around the resizer if there are any scrollbars present.
+    if (!hasOverlayScrollbars() && (m_vBar || m_hBar) && renderer.style().scrollbarWidth() != ScrollbarWidth::None)
+        RenderTheme::singleton().paintPlatformResizerFrame(renderer, context, resizerAbsRect);
```
```suggestion
       renderer.theme().paintPlatformResizerFrame(renderer, context, resizerAbsRect);
```

aprotyas commented on Source/WebCore/rendering/RenderTheme.cpp:
```diff
@@ -1462,6 +1462,53 @@ void RenderTheme::paintSliderTicks(const RenderObject& renderer, const PaintInfo
+    }
+
+    if (renderer.shouldPlaceVerticalScrollbarOnLeft()) {
+        context.save();
```
Nit: GraphicsContextStateSaver?

aprotyas commented on Source/WebCore/rendering/RenderTheme.cpp:
```diff
@@ -1462,6 +1462,53 @@ void RenderTheme::paintSliderTicks(const RenderObject& renderer, const PaintInfo
+
+void RenderTheme::paintPlatformResizerFrame(RenderLayerModelObject& renderer, GraphicsContext& context, const LayoutRect& resizerAbsRect)
+{
+    UNUSED_PARAM(renderer);
```
Alternatively remove the name of the param
```suggestion
void RenderTheme::paintPlatformResizerFrame(RenderLayerModelObject&, GraphicsContext& context, const LayoutRect& resizerAbsRect)
{
```

pxlcoder replied:
```suggestion
void RenderTheme::paintPlatformResizerFrame(RenderLayerModelObject&, GraphicsContext& context, const LayoutRect& resizerAbsRect)
{
```

lilyspiniolas committed:
[macOS] Add vector-based version of resizer
https://bugs.webkit.org/show_bug.cgi?id=289806
rdar://147062251

Reviewed by Abrar Rahman Protyasha and Aditya Keerthi.

Added support for a vector-based version of the resizer. Moved
codepath for drawing the resizer image and resizer frame to RenderTheme
in order to account for platform differences.

* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::paintResizer):
(WebCore::RenderLayerScrollableArea::drawPlatformResizerImage): Deleted.
* Source/WebCore/rendering/RenderLayerScrollableArea.h:
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::paintPlatformResizer):
(WebCore::RenderTheme::paintPlatformResizerFrame):
* Source/WebCore/rendering/RenderTheme.h:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.h:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::RenderThemeCocoa::paintPlatformResizer):
(WebCore::RenderThemeCocoa::paintPlatformResizerFrame):

Canonical link: https://commits.webkit.org/292298@main
